core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -304,6 +304,9 @@ public QueryBuilder postFilter() {
      * From index to start the search from. Defaults to <tt>0</tt>.
      */
     public SearchSourceBuilder from(int from) {
+        if (from < 0) {
+            throw new IllegalArgumentException("from must be no negative but was [" + from + "]");
```

Comment:
A better wording for the error is: `[from] can't be negative`
Changing that will require updating tests